### ISSUE TITLE
同样是用 CPU 进行的矩阵运算， torch 要比 paddle 快 300 倍

### ISSUE THREAD

**Issue by fnhweiqs** [PaddlePaddle/Paddle#45323]
#### z.shape = [8, 3328, 320]
s = time.time()
sim = paddle.matmul(z, z.transpose(perm=[0, 2, 1]))
logits = paddle.tril(sim, -1)[:, :, :-1]
logits += paddle.triu(sim, 1)[:, :, 1:]
logits = -F.log_softmax(logits, axis=-1)
e = time.time()
print(e - s)
>> 59.606117963790894

#### z.shape = [8, 3328, 320]
s = time.time()
sim = torch.matmul(z, z.transpose(1, 2))
logits = torch.tril(sim, -1)[:, :, :-1]
logits += torch.triu(sim, 1)[:, :, 1:]
logits = -F.log_softmax(logits, dim=-1)
e = time.time()
print(e - s)
>> 0.17826390266418457

**Comment by lyuwenyu** [comment]
我测了下时间主要是在log_softmax这个OP  @juncaipeng  看下

**Comment by yuwoyizhan** [comment]
性能差距这么大吗?我本人对性能最感兴趣,那希望飞桨好好测试吧.以前也看过一个issue,说的是飞桨动态图小神经网络训练比pytorch慢几倍.飞桨是不是在小神经网络上优化不到位呢?只注重应用的模型呢?应用的模型往往比较大,几十层几百层神经网络.比如resnet,文心大模型.

**Comment by Ligoml** [comment]
辛苦 @juncaipeng 修复一下这个算子性能问题，另外 @yuwoyizhan 如果你对算子性能优化感兴趣，欢迎参与飞桨框架的算子性能优化贡献 https://github.com/PaddlePaddle/Paddle/issues/42286
